File: __tests__/fixtures/vuln/hardcoded_secret.ts
// VULNERABILITY FIXTURE - for testing gitleaks / SAST detection
const API_KEY = 'sk-proj-abcdefghijklmnopqrstuvwxyz1234567890ABCD'
const DB_PASSWORD = 'super_secret_password_123!'
const STRIPE_KEY = 'FAKE_STRIPE_KEY_FOR_GITLEAKS_TESTING_ONLY'

export function connectToService() {
  return fetch('https://api.example.com', {
    headers: { 'X-API-Key': API_KEY }
  })
}

File: __tests__/fixtures/vuln/hardcoded_secret.java
// VULNERABILITY FIXTURE - for testing gitleaks / SAST detection
public class VulnConfig {
    private static final String DB_PASSWORD = "P@ssw0rd!Database123";
    private static final String API_KEY = "sk-test-abcdefghij1234567890";

    public static String getPassword() {
        return DB_PASSWORD;
    }
}
